crates/revm/revm-primitives/src/env.rs

```diff
@@ -42,9 +42,14 @@ pub fn fill_cfg_env(
     cfg_env.perf_analyse_created_bytecodes = AnalysisKind::Analyse;
 }
 /// Fill block environment from Block.
-pub fn fill_block_env(block_env: &mut BlockEnv, header: &Header, after_merge: bool) {
+pub fn fill_block_env(
+    block_env: &mut BlockEnv,
+    chain_spec: &ChainSpec,
+    header: &Header,
+    after_merge: bool,
+) {
     block_env.number = U256::from(header.number);
-    block_env.coinbase = header.beneficiary;
+    block_env.coinbase = dbg!(block_coinbase(chain_spec, header, after_merge));
```

Comment:
block_env.coinbase = dbg!(block_coinbase(chain_spec, header, after_merge));
```suggestion
    block_env.coinbase = block_coinbase(chain_spec, header, after_merge);
```